pkg/model/stack_serializer.go

```diff
@@ -317,22 +317,27 @@ func (s *Stack) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
-func unmarshalVolume(volume *VolumeTopLevel) (*VolumeSpec, error) {
-
-	result := &VolumeSpec{}
-	if volume == nil {
-		result.Labels = make(Labels)
-		result.Annotations = make(Annotations)
-	} else {
-		result.Labels = make(Labels)
-		if volume.Annotations == nil {
-			result.Annotations = make(Annotations)
-		} else {
-			result.Annotations = volume.Annotations
-		}
+func unmarshalVolume(volume *VolumeTopLevel, isCompose bool) (*VolumeSpec, error) {
+	result := &VolumeSpec{
+		Labels:      make(Labels),
+		Annotations: make(Annotations),
+	}
 
-		for key, value := range volume.Labels {
-			result.Annotations[key] = value
+	if volume != nil {
```

Comment:
NIT: Maybe we can do something like this in order tohave less indentation and make it more readable. WDYT?
```suggestion
	if volume == nil {
	    return result
	}
	if isCompose {
	...
```

Reply:
I don't have a strong preference so I've updated it in commit: fcc5bc75aad81c54cf7b5d1500a6907ddba4e2ef